control/plugin_manager.go

```diff
@@ -180,51 +180,51 @@ type loadedPlugin struct {
 
 // Name returns plugin name
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) Name() string {
+func (lp loadedPlugin) Name() string {
 	return lp.Meta.Name
 }
 
 // PluginPath returns the plugin path
-func (lp *loadedPlugin) PluginPath() string {
+func (lp loadedPlugin) PluginPath() string {
 	return lp.Details.Path
 }
 
 // Key returns plugin type, name and version
-func (lp *loadedPlugin) Key() string {
+func (lp loadedPlugin) Key() string {
 	return fmt.Sprintf("%s"+core.Separator+"%s"+core.Separator+"%d", lp.TypeName(), lp.Name(), lp.Version())
 }
 
 // Version returns plugin version
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) Version() int {
+func (lp loadedPlugin) Version() int {
 	return lp.Meta.Version
 }
 
 // TypeName returns plugin type as a string
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) TypeName() string {
+func (lp loadedPlugin) TypeName() string {
 	return lp.Type.String()
 }
 
 // Status returns current plugin state
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) Status() string {
+func (lp loadedPlugin) Status() string {
 	return string(lp.State)
 }
 
 // IsSigned returns plugin signing as a bool
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) IsSigned() bool {
+func (lp loadedPlugin) IsSigned() bool {
 	return lp.Details.Signed
 }
 
 // LoadedTimestamp returns a unix timestamp of the LoadTime of a plugin
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) LoadedTimestamp() *time.Time {
+func (lp loadedPlugin) LoadedTimestamp() *time.Time {
 	return &lp.LoadedTime
 }
 
-func (lp *loadedPlugin) Policy() *cpolicy.ConfigPolicy {
+func (lp loadedPlugin) Policy() *cpolicy.ConfigPolicy {
```

Comment:
> In either case you're going to need to copy the individual elements as well because loadedPlugin has reference elements.

Here I was talking about cpolicy/plugin details being pointers not about how receivers work.

>All these methods are 'getters', so they should only be used to retrieve values, so it should not be the case.

My comment wasn't really directed at this but getter methods that return pointers still allow you to mutate the object, even though the receiver is a value receiver.

I was trying to bring up that changing loadedPlugin from pointer receivers to value receivers could result in weird inconsistencies since there are still places there use loadedPlugin as a pointer. Could be confusing in the future.